test/hotspot/jtreg/serviceability/jvmti/SuspendWithObjectMonitorWait/SuspendWithObjectMonitorWait1.java

```diff
@@ -0,0 +1,223 @@
+            // set TS_WAITER_RUNNING and notified barrierLaunch above so
+            // we cannot enter threadLock until the waiter thread calls
+            // threadLock.wait().
+            synchronized (threadLock) {
+                // notify waiter thread so it can try to reenter threadLock
+                testState = TS_READY_TO_NOTIFY;
+                threadLock.notify();
+
+                // wait for the waiter thread to block
+                logDebug("before contended enter wait");
+                int retCode = wait4ContendedEnter(waiter);
+                if (retCode != 0) {
+                    throw new RuntimeException("error in JVMTI GetThreadState: "
+                                               + "retCode=" + retCode);
+                }
+                logDebug("done contended enter wait");
+
+                checkTestState(TS_READY_TO_NOTIFY);
+                testState = TS_CALL_SUSPEND;
+                logDebug("before suspend thread");
+                retCode = suspendThread(waiter);
+                if (retCode != 0) {
+                    throw new RuntimeException("error in JVMTI SuspendThread: "
+                                               + "retCode=" + retCode);
+                }
+                logDebug("suspended thread");
+            }
+
+            //
+            // At this point, all of the child threads are running
+            // and we can get to meat of the test:
+            //
+            // - suspended threadLock waiter (trying to reenter)
+            // - a threadLock enter in the resumer thread
+            // - resumption of the waiter thread
+            // - a threadLock enter in the freshly resumed waiter thread
+            //
+
+            synchronized (barrierResumer) {
+                checkTestState(TS_CALL_SUSPEND);
+
+                // tell resumer thread to resume waiter thread
+                testState = TS_READY_TO_RESUME;
+                barrierResumer.notify();
+
+                // Can't call checkTestState() here because the
+                // resumer thread may have already resumed the
+                // waiter thread.
+            }
+
+            try {
+                resumer.join(JOIN_MAX * 1000);
+                if (resumer.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("resumer thread is stuck");
+                }
+                waiter.join(JOIN_MAX * 1000);
+                if (waiter.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("waiter thread is stuck");
+                }
+            } catch (InterruptedException ex) {
+            }
+
+            checkTestState(TS_WAITER_DONE);
+        }
+
+        System.out.println("Executed " + count + " loops in " + timeMax +
+                           " seconds.");
+
+        return 0;
+    }
```

Comment:
Nit: Launch of `Waiter` and `Resumer`, some testing steps and test shutdown are identical for all 3 tests. So, they can be factored out into separate functions located in the base file.